app/src/lib/stores/app-store.ts

```diff
@@ -3037,6 +3000,50 @@ export class AppStore extends TypedBaseStore<IAppState> {
     return repository
   }
 
+  private async _stashBasedOnUncommittedChangesStrategy(
+    repository: Repository,
+    branch: Branch,
+    uncommittedChangesStrategy: UncommittedChangesStrategy = askToStash
+  ): Promise<IStashEntry | null> {
+    const {
+      changesState,
+      branchesState: { tip },
```

Comment:
✨oooh nested destructuring... I forget about this

Reply:
I didn't know if this was too weird. It's kind of hard to follow and spitting it into two lines might be more readable.